driver/controller.go

```diff
@@ -177,11 +177,90 @@ func (d *Driver) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest)
 	ll.Info("delete volume called")
 
 	resp, err := d.doClient.Storage.DeleteVolume(ctx, req.VolumeId)
+	if err == nil {
+		ll.WithField("response", resp).Info("volume is deleted")
+		return &csi.DeleteVolumeResponse{}, nil
+	}
+
+	// there are no ways to intepret the error, return it
+	if resp == nil {
+		return nil, err
+	}
+
+	// we assume it's deleted already for idempotency
+	if resp.StatusCode == http.StatusNotFound {
```

Comment:
log here

Reply:
This also happens if the plugin receives two or more delete calls. The first one deletes it, the second one gets a `StatusNotFound`.